File: frontend/src/services/sessionService.ts
import { SESSION_PAGE_SIZE } from "@/domain/constants"
import { getSessionLabel } from "@/domain/session"
import type { Session } from "@/domain/entities/chat"
import { sessionRepository } from "@/repositories/sessionRepository"
import { storageService } from "@/services/storageService"

export async function listSessions(limit = SESSION_PAGE_SIZE, offset = 0) {
  return sessionRepository.fetchSessions(limit, offset)
}

export async function renameConversation(
  sessionId: string,
  name: string,
): Promise<string> {
  const trimmedName = name.trim()
  if (!trimmedName || trimmedName.length > 256) {
    return ""
  }

  await sessionRepository.renameSession(sessionId, trimmedName)
  return trimmedName
}

export async function deleteConversation(sessionId: string): Promise<void> {
  await sessionRepository.deleteSession(sessionId)
}

export async function updateConversationSystemPrompt(
  sessionId: string,
  content: string,
): Promise<void> {
  await sessionRepository.updateSystemPrompt(sessionId, content)
}

export function getConversationLabel(session: Session): string {
  return getSessionLabel(session)
}

export function getStoredSidebarState(): boolean {
  return storageService.getSidebarCollapsed()
}

export function setStoredSidebarState(collapsed: boolean): void {
  storageService.setSidebarCollapsed(collapsed)
}

File: ThreadGPTiOSV2/ThreadGPTiOSV2/Application/SessionService.swift
import Foundation

final class SessionService {
    private let repository: SessionRepository

    init(repository: SessionRepository) {
        self.repository = repository
    }

    func listSessions(limit: Int = ThreadGPTConstants.sessionPageSize, offset: Int = 0) async throws -> SessionListPage {
        try await repository.fetchSessions(limit: limit, offset: offset)
    }

    func renameConversation(sessionID: String, name: String) async throws -> String {
        let trimmed = name.trimmingCharacters(in: .whitespacesAndNewlines)
        guard !trimmed.isEmpty, trimmed.count <= 256 else { return "" }
        try await repository.renameSession(sessionID: sessionID, name: trimmed)
        return trimmed
    }

    func deleteConversation(sessionID: String) async throws {
        try await repository.deleteSession(sessionID: sessionID)
    }

    func updateConversationSystemPrompt(sessionID: String, content: String) async throws {
        try await repository.updateSystemPrompt(sessionID: sessionID, systemPrompt: content)
    }
}

